cgi-bin/LJ/Widget/ShopCart.pm

```diff
@@ -26,241 +26,93 @@ sub need_res { qw( stc/shop.css ) }
 
 sub render_body {
     my ( $class, %opts ) = @_;
-
-    my $ret;
-
     my $cart = $opts{cart} ||= DW::Shop->get->cart
         or return $class->ml('widget.shopcart.error.nocart');
 
     return $class->ml('widget.shopcart.error.noitems')
         unless $cart->has_items;
 
+    my $receipt = $opts{receipt};
+
     # if the cart is not in state OPEN, mark this as a receipt load
     # no matter where we are
-    $opts{receipt} = 1
+    $receipt = 1
         unless $cart->state == $DW::Shop::STATE_OPEN;
-    $opts{receipt} = 1
+    $receipt = 1
         if $opts{admin};
-    $opts{receipt} = 1
+    $receipt = 1
         if $opts{confirm};
 
     # if we're not doing a receipt load, then we should balance the points.  this
     # fixes situations where the user gets gifted points while they're shopping.
-    unless ( $opts{receipt} ) {
+    unless ($receipt) {
         $cart->recalculate_costs;
         $cart->save;
     }
 
     my $colspan = $opts{receipt} ? 5 : 6;
```

Comment:
Did you also mean to change this from `$opts{receipt}` to `$receipt`? If not, can you explain the difference in a comment?